compose/generic-x86.yml

```diff
@@ -6,11 +6,15 @@ services:
   core:
     build: ./core
     network_mode: 'host'
+    privileged: true
     volumes:
       - 'core-storage:/data'
       - 'reports-storage:/reports'
+      - '/var/run/docker.sock:/var/run/docker.sock'
+      - '/dev:/dev'
```

Comment:
This is redundant in combination with `privileged: true`.

Reply:
Also, if we know _what_ device(s) we need access to, we're better off being more explicit than less.

Reply:
But we are removing privileged, correct? :wink: 

What are the minimum caps required for unwrapping?

Reply:
@klutchell Looks like we need to start the container with `/dev/loop-control` mapped, add a cgroup rule for loopback device major numbers, and add the MKNOD capability:

`docker run -it --device /dev/loop-control --device-cgroup-rule="b 7:* rwm" --cap-add MKNOD alpine`

We can then create the `/dev/loop*` device files with the corresponding minors:

`for i in $(seq 0 9); do mknod -m 0600 /dev/loop${i} b 7 $i; done`

And use them like outside the container:

`fallocate -l 1G test.img && losetup -fP test.img`

Reply:
It should be noted that using this approach, we're using the same minor numbers as the host OS (and there's no way to isolate these), so the guest has access to map or modify mappings of the host, and any mappings inside the guest also show up on the host. This also means that we have the same limits as the host, and if we don't clean up, we'll soon run out of loop devices.

Reply:
@jakogut what would a cleanup look like? It's starting to look like the unwrap approach is going to introduce a lot of risk but the generic images are only publishing flasher types at the moment.

Reply:
As discussed in the call a little bit ago, I'm pursuing an approach reading the image directly from the filesystem at the partition offset using Node, which the Etcher project has already pioneered for us.

Reply:
@jakogut feel free to push to this PR and change the features/permissions exposed in the compose file as needed, locking it down as you do